roachtest: lib/pq failed

roachtest.lib/pq [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/9880122?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/9880122?buildTab=artifacts#/lib/pq) on release-23.1.0 @ [36ee6c802b18ab8a544b20d03dec68d82d99604b](https://github.com/cockroachdb/cockroach/commits/36ee6c802b18ab8a544b20d03dec68d82d99604b):


```
test artifacts and logs in: /artifacts/lib/pq/run_1
(orm_helpers.go:212).summarizeFailed: 
Tests run on Cockroach v23.1.0-rc.1-dev-36ee6c802b18ab8a544b20d03dec68d82d99604b
Tests run against lib/pq v1.10.5
217 Total Tests Run
189 tests passed
28 tests failed
7 tests skipped
0 tests ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: pq.TestContextCancelExec - unknown (unexpected)
For a full summary look at the lib/pq artifacts
An updated blocklist (libPQBlocklist) is available in the artifacts' lib/pq log
```
<p>Parameters: <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=4</code>
, <code>ROACHTEST_encrypted=false</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
/cc @cockroachdb/sql-sessions
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*lib/pq.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-27584

--- a/pkg/cmd/roachtest/tests/libpq_blocklist.go
+++ b/pkg/cmd/roachtest/tests/libpq_blocklist.go
@@ -53,4 +53,8 @@ var libPQIgnorelist = blocklist{
 	// pq: only text format supported for COPY, however no error is returned
 	// for CRDB.
 	"TestCopyInBinaryError": "63235",
+	// TestContextCancelExec has a race between context cancellation and query
+	// execution.
+	// https://github.com/lib/pq/blob/381d253611d666974d43dfa634d29fe16ea9e293/go18_test.go#L92
+	"TestContextCancelExec": "102674",
 }

